### PR TITLE
Test and refactor Homestays controller

### PR DESCRIPTION
**Pull request by rbmrclo**
#### TODO:
- [x] Refactor Homestays Controller
- [x] Test Homestays Controller
#### Notes:
- This PR is pointing to #11, since there's some changes that I heavily rely from that PR.
- ~~Analytics [has been removed](https://github.com/pethomestay/phs/pull/15/files#diff-3e28378ed277b57b1abfcd9578fd469aL58) from HomestaysController#show since we're deciding to unsupport `KISSmetrics` and `Google Analytics` integration isn't used anyway.~~

From Segment.io docs:
screen shot 2015-09-30 at 1 55 27 pm
#### Changelog:
- Deleted `Search#perform` https://github.com/pethomestay/phs/commit/1f3cf7b446b05e5265839a14ba3023d620da18ac
- Refactor/Test Homestays Controller https://github.com/pethomestay/phs/commit/2588603a49d05dd1614db5f2e4d70129d4494c19, https://github.com/pethomestay/phs/commit/b7fcaaa031834736a1ea445a812f4deac82dfc24, https://github.com/pethomestay/phs/commit/627b1ae129627efeac92ade6108dc2e2fb1f83e0
- Expose `HomestaySearch#search` object https://github.com/pethomestay/phs/commit/30f4d3a3ef87b4305f4be08400c4bfb1cb584989
- Updates factories https://github.com/pethomestay/phs/commit/90a2f5eed348c0e32721988fb13b7d0b1003b841
#### Missing
- [x] ~~[HomestaysController#availability](https://github.com/pethomestay/phs/pull/15/files#diff-3e28378ed277b57b1abfcd9578fd469aR72) test~~
- [ ] Refactor Analytics track

@zero51 @adimasuhid Please review. Thanks!

### CODE REVIEW

**Comment by rbmrclo** [comment]
I just realized that this [method](https://github.com/pethomestay/phs/pull/15/files#diff-3e28378ed277b57b1abfcd9578fd469aL109) has been deleted [here](https://github.com/pethomestay/phs/pull/9/files#diff-4676c008b11a5480d73d4a6de01e45b9L394) already (depreciated).

**Comment by rbmrclo** [comment]
I'm currently trying out `segment.io` (registered via personal account) and It seems like I overlooked how `Analytics.track` works. I should have not removed that code. (I find it very similar with mixpanel's `track` method that's why I'm very skeptical if I should remove it or not.)

Will revert the code with a touch of refactor. My bad.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def track_session_variables
           'UserAgent' => request.user_agent,
           'ip' => request.ip,
         },
-        integrations:  { 'Google Analytics' => false, 'KISSmetrics' => true } 
+        integrations:  { 'Google Analytics' => false, 'KISSmetrics' => true }
```
Comment (zero51) [inline app/controllers/application_controller.rb:88]: So we still need this, despite the fact that they don't use kiss metrics?

File: app/controllers/homestays_controller.rb [hunk 2]
```diff
@@ -1,112 +1,102 @@
...
-    if @homestay.locked?
-      notice = nil #we will change this to a message later
+      @hs = HomestaySearch.new(params[:search])
+      @hs.perform_search
+
+      @search    = @hs.search
+      @homestays = @hs.results
+      @homestays = @homestays.paginate(page: params[:page], per_page: 10)
+
+      gon.push({ search: @search, homestays: @homestays })
```
Comment (zero51) [inline app/controllers/homestays_controller.rb:23]: It's weird how we're pushing all the homestay data to the client twice: once in JS and once with HTML.

**Comment by zero51** [comment]
Looks good to me.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (rbmrclo) [inline app/controllers/application_controller.rb:88]: I played around with it's API and it seems like i'm still able to trigger an event, even without those integrations. It seems like this is very important, so yes, we still need this.

screen shot 2015-10-10 at 9 04 09 am

screen shot 2015-10-10 at 9 04 00 am

**Comment by rbmrclo** [comment]
Moved to: https://github.com/pethomestay/phs/pull/22

**Pull request closed without merging**